UserTools/DirectionGridCheck/DirectionGridCheck.cpp

```diff
@@ -0,0 +1,245 @@
+    double RecoDirY = recovtx->GetDirection().Y();
+    double RecoDirZ = recovtx->GetDirection().Z();
+    double recoPhi = atan(RecoDirY/RecoDirX);
+    double recoTheta = asin(RecoDirX * RecoDirX + RecoDirY * RecoDirY);
+    if (trueTheta < 0) trueTheta += 2*TMath::Pi();
+    if (trueTheta > 2 * TMath::Pi()) trueTheta -= 2 * TMath::Pi();
+    TrueDirPlot->SetPoint(eventCount, truePhi, trueTheta);
+    std::cout << "True Plot set: " << eventCount << ", " << truePhi << ", " << trueTheta << endl;
+    Direction MRDDirection = this->findDirectionMRD();
+    MRDDirPlot->SetPoint(eventCount, MRDDirection.GetPhi(), MRDDirection.GetTheta());
+    std::cout << "MRD Plot set: " << eventCount << ", " << MRDDirection.GetPhi() << ", " << MRDDirection.GetTheta() << endl;
+    RecoDirPlot->SetPoint(eventCount, recoPhi, recoTheta);
+    std::cout << "recoPlot set: " << eventCount << ", " << recoPhi << ", " << recoTheta << endl;
+
+    eventCount++;
+  return true;
+}
+
+
+bool DirectionGridCheck::Finalise(){
+    fOutput_tfile->cd();
+    AngularPlot->Write("Angular Grid");
+    TrueDirPlot->Write("trueDirection");
+    MRDDirPlot->Write("MRDDirection");
+    RecoDirPlot->Write("recoDirection");
+    fOutput_tfile->Write();
+    fOutput_tfile->Close();
+  return true;
+}
+
+Direction DirectionGridCheck::findDirectionMRD() {
+	std::vector<BoostStore>* Tracks;
+    int numtracksinev;
+	m_data->Stores["MRDTracks"]->Get("MRDTracks", Tracks);
+	m_data->Stores["MRDTracks"]->Get("NumMrdTracks", numtracksinev);
+
+	if (numtracksinev > 1) Log("Multiple tracks need work; just using first for now", v_debug, verbosity);
+	double gradx, grady, theta, phi;
+	Direction startVertex, endVertex, result;
+	BoostStore* thisTrack = &(Tracks->at(0));
+
+	thisTrack->Get("VTrackGradient", gradx);
+	thisTrack->Get("HTrackGradient", grady);
+	phi = atan(grady / gradx);
+	theta = asin(pow((gradx * gradx + grady * grady), 0.5));
+	/*TRandom3 smear;
+	Direction vtxDir = fTrueVertex->GetDirection();
+	Direction result;
+	result.SetTheta(smear.Gaus(vtxDir.GetTheta(), 0.4));
+	result.SetPhi(smear.Gaus(vtxDir.GetPhi(), 0.4));*/
+	result.SetTheta(theta);
+	result.SetPhi(phi);
+
+	return result;
+}
+
+RecoVertex* DirectionGridCheck::FindSimpleDirection(RecoVertex* myVertex) {
+
+    /// get vertex position
+    double vtxX = myVertex->GetPosition().X();
+    double vtxY = myVertex->GetPosition().Y();
+    double vtxZ = myVertex->GetPosition().Z();
+    double vtxTime = myVertex->GetTime();
+
+    std::cout << "Simple Direction Input Position: (" << vtxX << "," << vtxY << "," << vtxZ << ")\n";
+    // current status
+    // ==============
+    int status = myVertex->GetStatus();
+
+    /// loop over digits
+    /// ================
+    double Swx = 0.0;
+    double Swy = 0.0;
+    double Swz = 0.0;
+    double Sw = 0.0;
+    double digitq = 0.;
+    double dx, dy, dz, ds, px, py, pz, q;
+
+    RecoDigit digit;
+    for (int idigit = 0; idigit < fDigitList->size(); idigit++) {
```

Comment:
this is the first and only reference to `fDigitList` - the variable is initialised to a nullptr in the header but never set, so this will segfault.